aws-synthetics-canary/aws-synthetics-canary.json

```diff
@@ -0,0 +1,200 @@
+            "type" : "object",
+            "additionalProperties" : false,
+            "properties" : {
+                "Expression" : {
+                    "type" : "string",
+                    "pattern": "^rate(.*.)$"
+                },
+                "DurationInSeconds" : {
+                    "type" : "string"
+                }
+            },
+            "required" : [ "Expression", "DurationInSeconds" ]
+        },
+        "Code" : {
+            "type" : "object",
+            "additionalProperties" : false,
+            "properties" : {
+                "S3Bucket": {
+                    "type": "string"
+                },
+                "S3Key": {
+                    "type": "string"
+                },
+                "S3ObjectVersion": {
+                    "type": "string"
+                },
+                "Script": {
+                    "type": "string"
+                },
+                "Handler": {
+                    "type": "string"
+                }
+            },
+            "oneOf": [
+                {
+                    "required": [
+                        "S3Bucket",
+                        "S3Key",
+                        "Handler"
+                    ]
+                },
+                {
+                    "required": [
+                        "Script",
+                        "Handler"
+                    ]
+                }
+            ]
+        },
+        "Tag" : {
+            "type" : "object",
+            "additionalProperties" : false,
+            "properties" : {
+                "Key" : {
+                    "type" : "string"
+                },
+                "Value" : {
+                    "type" : "string"
+                }
+            },
+            "required" : [ "Value", "Key" ]
+        },
+        "VpcConfig" : {
+            "type" : "object",
+            "additionalProperties" : false,
+            "properties" : {
+                "VpcId" : {
+                    "type" : "string"
+                },
+                "SubnetIds" : {
+                    "type" : "array",
+                    "pattern": "^([0-9a-z_\\-]{0-16})$",
+                    "items" : {
+                        "type" : "string"
+                    }
+                },
+                "SecurityGroupIds" : {
+                    "type" : "array",
+                    "pattern": "^([0-9a-z_\\-]{0-5})$",
+                    "items" : {
+                        "type" : "string"
+                    }
+                }
+            },
+            "required" : [ "VpcId", "SubnetIds", "SecurityGroupIds" ]
+        }
+    },
+    "required": [
+        "Name",
+        "Code",
+        "ArtifactLocation",
+        "ExecutionRoleArn",
+        "Schedule",
+        "RuntimeVersion",
+        "StartCanaryAfterCreation"
+    ],
+    "handlers": {
+        "create": {
+            "permissions": [
+                "s3:CreateBucket",
+                "s3:PutBucketEncryption",
```

Comment:
think the IAM action for this API is different:
https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazons3.html